### PR TITLE
Fixed observables for source-manager

### PR DESCRIPTION
**Pull request by aklochkova**
Fixes https://github.com/OHDSI/Atlas/issues/1101

### CODE REVIEW

File: js/modules/WebAPIProvider/VocabularyProvider.js [hunk 1]
```diff
@@ -35,14 +35,16 @@ define(function (require, exports) {
...
 		// preload domain list once for all future calls to getDomains()
-		$.ajax({
-			url: config.webAPIRoot + 'vocabulary/' + defaultSource.sourceKey + '/domains',
-		}).then(function (results) {
-			$.each(results, function (i, v) {
-				domains.push(v.DOMAIN_ID);
-			});
-			domainPromise.resolve(domains);
-		});
+		if (defaultSource !== undefined){
```
Comment (pavgra) [inline js/modules/WebAPIProvider/VocabularyProvider.js]: What happens if we start Atlas w/o sources and add a source via UI? The default vocab is not loaded?

**Comment by chrisknoll** [comment]
Sorry, question: do we need to have getX and getXPromise in this module API?  Seems stange.  we understand that these service APIs are synchronous, so getDomains should return a promise.  Could we not reshuffle the code like this?

File: js/modules/WebAPIProvider/VocabularyProvider.js [hunk 2]
```diff
@@ -118,7 +124,7 @@ define(function (require, exports) {
 		return deferred.promise();
 	}
 
-	function getDomains() {
+	function getDomainPromise() {
```
Comment (chrisknoll) [inline js/modules/WebAPIProvider/VocabularyProvider.js]: Just need to revert this rename and just have 'getDomains' exposed int he api below.

File: js/modules/WebAPIProvider/VocabularyProvider.js [hunk 3]
```diff
@@ -13,37 +13,43 @@ define(function (require, exports) {
...
-		});
-	})
+	
+	getDomains();
+	
+	function getDomains() {
+        sourceAPI.getSources().then(function (sources) {
+            if (sources.length == 0) {
+                sharedState.appInitializationStatus('no-sources-available');
+                return;
```
Comment (chrisknoll) [inline js/modules/WebAPIProvider/VocabularyProvider.js]: I believe at this point, when we have no sources, we can resolve the domains promise to either [] or null.  Return [] if domains is expected to return a list.

**Comment by chrisknoll** [comment]
So, I have some additional updates to this I'd like to make but realized that this is for a 2.4.5 branch, and I'm only really interested in updates to master.  Is there a separate PR for this for the master branch, or is that coming later?

**Pull request merged**